### ISSUE TITLE
Event is missing in system projection

### ISSUE THREAD

**Issue by jvdlaar** [EventStore/EventStore#1877]
We have an event that has been created in the eventstore, but doesn't seem to end up in the system category or event type projections. This is the payload:

```
{
  "eventId": "7fd6b3a9-d7bb-4ca8-90a2-601c70d22f9b",
  "eventType": "ImportStarted",
  "eventNumber": 1,
  "data": "{\n  \"import_id\": \"7e04b8c9-91ff-415e-a96a-692a18d12ee8\",\n  \"feed_id\": \"3c7f052c-b51d-5834-b44d-01faac941d7e\",\n  \"date\": \"2019-02-21T12:40:03+00:00\",\n  \"recurring\": true\n}",
  "metaData": "{\n  \"_aggregate_id\": \"7e04b8c9-91ff-415e-a96a-692a18d12ee8\",\n  \"_aggregate_version\": 2,\n  \"_aggregate_type\": \"Domain\\\\Importing\\\\Import\",\n  \"$causationId\": \"ea35ceda-f003-48ba-8b8e-b0eb8d1605ed\",\n  \"$correlationId\": \"f6de51a0-bf53-4fb1-ba2a-5ee7e2a8d24c\"\n}",
  "streamId": "ImportV2-7e04b8c991ff415ea96a692a18d12ee8",
  "isJson": true,
  "isMetaData": true,
  "isLinkMetaData": false,
  "positionEventNumber": 1,
  "positionStreamId": "ImportV2-7e04b8c991ff415ea96a692a18d12ee8",
  "title": "1@ImportV2-7e04b8c991ff415ea96a692a18d12ee8",
  "id": "URL/streams/ImportV2-7e04b8c991ff415ea96a692a18d12ee8/1",
  "updated": "2019-02-21T12:40:03.435996Z",
  "author": {
    "name": "EventStore"
  },
  "summary": "ImportStarted",
  "links": [
    {
      "uri": "URL/streams/ImportV2-7e04b8c991ff415ea96a692a18d12ee8/1",
      "relation": "edit"
    },
    {
      "uri": "URL/streams/ImportV2-7e04b8c991ff415ea96a692a18d12ee8/1",
      "relation": "alternate"
    }
  ]
}
```

So if I look in this stream: /streams/$ce-ImportV2 or /streams/$et-ItemStarted then the event is not there. If I create a new stream fromStream('ImportV2-7e04b8c991ff415ea96a692a18d12ee8'), then the event does exist in the new projection stream.

We're running eventstore 5.0.0 (at the time this event was created it might have been one of the rc's). And we run this in a kubernetes cluster / docker containers.

If I look in the logs around the time that the event is created then I see a lot of these:
```
Error while processing message EventStore.Projections.Core.Messaging.UnwrapEnvelopeMessage in queued handler '"Projection Core #1"'.
EXCEPTION OCCURRED
Invalid JSON
```

But it shows a lot of such messages for about 2 minutes (not just one or a couple), and then doesn't show such messages for a couple for days. And then it returns again for a couple of minutes.

It appears to be that these errors start occuring after
![screenshot from 2019-02-27 09-17-02](https://user-images.githubusercontent.com/1094732/53478893-2c959d80-3a78-11e9-84d4-24f653da0345.png)
![screenshot from 2019-02-27 10-14-15](https://user-images.githubusercontent.com/1094732/53479035-7c746480-3a78-11e9-94d4-c26952d44730.png)

**Comment by shaan1337** [comment]
@jvdlaar hello, can you please send us the contents of the `$$ImportV2-7e04b8c991ff415ea96a692a18d12ee8` stream in your db?

It looks like the projection is failing when trying to deserialize the metadata stream.

**Comment by jvdlaar** [comment]
I do get a ImportV2-7e04b8c991ff415ea96a692a18d12ee8 stream, but the metadata stream seems to be gone.
![Screenshot from 2019-03-15 09-49-47](https://user-images.githubusercontent.com/1094732/54419467-d1b2a600-4707-11e9-94f5-029f5aea85ae.png)
![Screenshot from 2019-03-15 09-50-43](https://user-images.githubusercontent.com/1094732/54419468-d24b3c80-4707-11e9-9164-f3ad797cd284.png)

**Comment by shaan1337** [comment]
@jvdlaar thank you. have you been modifying the metadata of any streams (particularly system streams beginning with `$`) in your code? The `Invalid JSON` error seems to occur because the `$by_category` projection assumes that system streams will have JSON metadata but when it is not the case, it throws an "Invalid JSON" exception.

**Comment by jvdlaar** [comment]
@shaan1337 I don't think we did. But the config does show a lot of red boxes. But all our projections seem to do that, I don't think the interface is properly loading the saved config.
![Screenshot from 2019-03-20 09-11-24](https://user-images.githubusercontent.com/1094732/54668880-46168c00-4af0-11e9-935a-a8d2157e6287.png)
![Screenshot from 2019-03-20 09-11-32](https://user-images.githubusercontent.com/1094732/54668881-46168c00-4af0-11e9-99f7-0e6bad66fa6f.png)

**Comment by oskardudycz** [comment]
@hayley-jean, could you check if we didn't fix that in the latest database versions? Thanks!
